content/200-concepts/100-components/02-prisma-client/260-client-extensions/120-query.mdx

```diff
@@ -147,6 +147,33 @@ const xprisma = prisma.$extends({
 })
 ```
 
+### Mutate the result of a query
+
+You can use `await` and then mutate the result of the `query` promise.
+
+```ts
+const xprisma = prisma.$extends({
+  query: {
+    user: {
+      async findFirst({ model, operation, args, query }) {
+        const user = await query(args)
+        
+        if (user.password !== undefined) {
+          user.password = '******'
+        }
+
+        return user
+      },
+    },
+  },
+})
+```
+
+
+<Admonition type="info">
+We include the above example to show that this is possible. However, for performance reasons we recommend that you use the [`result` component type](/concepts/components/prisma-client/client-extensions/result) to override existing fields.
```

Comment:
```suggestion
We include the above example to show that this is possible. However, we recommend that you use the [`result` component type](/concepts/components/prisma-client/client-extensions/result) to override existing fields. This is because in most cases performance will be better because the `result` component type only computes on access, but the `query` component type computer after the query is executed.
```